projects/rocprofiler-sdk/source/lib/output/generateRocpd.cpp

```diff
@@ -1391,6 +1392,269 @@ write_rocpd(
+                            ar(cereal::make_nvp("error_code", record.error_code));
+                        });
+
+                        break;
+                    }
+
+                    case ROCPROFILER_BUFFER_TRACING_KFD_PAGE_FAULT: {
+                        auto record = itr.page_fault_record;
+
+                        name  = tool_metadata.buffer_names.at(itr.kind, record.operation);
+                        tid   = record.pid; // KFD attributes all events to the thread_group's lead thread
+                        start = record.start_timestamp;
+                        end   = record.end_timestamp;
+                        value = record.address.value;
+
+                        json_data = get_json_string([&tool_metadata, &record](auto& ar) {
+                            ar(cereal::make_nvp("address", record.address.value));
+                            ar(cereal::make_nvp("agent_id", record.agent_id.handle == 0 ? -1 :
+                                                tool_metadata.get_agent(record.agent_id)->node_id));
+                        });
+
+                        break;
+                    }
+
+                    case ROCPROFILER_BUFFER_TRACING_KFD_QUEUE: {
+                        auto record = itr.queue_record;
+
+                        name  = tool_metadata.buffer_names.at(itr.kind, record.operation);
+                        tid   = record.pid; // KFD attributes all events to the thread_group's lead thread
+                        start = record.start_timestamp;
+                        end   = record.end_timestamp;
+
+                        json_data = get_json_string([&tool_metadata, &record](auto& ar) {
+                            ar(cereal::make_nvp("agent_id", record.agent_id.handle == 0 ? -1 :
+                                                tool_metadata.get_agent(record.agent_id)->node_id));
+                        });
+
+                        break;
+                    }
+
+                    default:
+                        continue;
+                    }
+
+                    auto name_id = string_entries.at(name);
+
+                    // insert thread info if it doesn't already exist
+                    get_thread_id(tid);
+
+                    // create an empty event, just to associate the various info tables
+                    auto evt_id = create_event(conn, {});
+
+                    // track timestamps with a track+sample for events, with a region for records
+                    if (start == end) {
+                        auto track_id = get_track_id(conn, node_id, this_pid, tid, name_id, "{}");
+                        auto sample_stmt = get_insert_statement("rocpd_sample{{uuid}}", {
+                            insert_value("track_id", track_id),
+                            insert_value("timestamp", start),
+                            insert_value("event_id", evt_id),
+                        });
+
+                        execute_raw_sql_statements(conn, sample_stmt);
```

Comment:
Maybe just create a region with the start == end; we already do this for roctxMark. I'd rather use the sample table for data that is truly sampled.